docs: update Kafka MSK IAM authentication configuration

kalavt opened the pull request:
- Replace deprecated aws_msk_iam and aws_msk_iam_cluster_arn parameters
- Add new rdkafka.sasl.mechanism=aws_msk_iam configuration method
- Add aws_region parameter for custom DNS/PrivateLink scenarios
- Update IAM permission examples for consumers and producers
- Add MSK Serverless example in input plugin
- Simplify documentation for user-friendly experience

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Rewrote and expanded AWS MSK IAM authentication guidance for Kafka, correcting plugin context to outputs and adopting a configuration-first flow.
  * Renamed/prioritized prerequisites, simplified examples, and introduced explicit parameters for SASL mechanism and region with auto-detection notes.
  * Added PrivateLink/custom DNS guidance with concrete examples and tabbed YAML/conf blocks.
  * Replaced broad IAM examples with scoped policies using REGION/ACCOUNT/CLUSTER/UUID placeholders and clarified credential discovery.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/2294)

<!-- review_stack_entry_end -->
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kalavt committed:
kafka: Add AWS MSK IAM auth support

Signed-off-by: Arbin <arbin.cheng@coins.ph>

kalavt committed:
docs: fix typo in Kafka output plugin description

Change 'Kafka input plugin' to 'Kafka output plugin' in the introduction

Signed-off-by: Arbin <arbin.cheng@coins.ph>

kalavt commented on pipeline/inputs/kafka.md:
````diff
@@ -136,73 +136,93 @@ The example can be executed locally with `make start` in the `examples/kafka_fil
-      match: '*'
+      rdkafka.sasl.mechanism: aws_msk_iam
+      aws_region: us-east-1
 ```
````
__`aws_msk_iam_cluster_arn`__: This parameter was __removed__ in this PR. Region now comes from `aws_region` parameter for custom DNS.

__`rdkafka.security.protocol`__: Auto-set by code (line 296-299 in `in_kafka.c`). Not needed in config.

Both parameters are intentionally excluded per PR design. Adding them contradicts the simplification goal.

alexakreizinger commented:
I want to give this my full attention, but I have to finish early for the day for an appointment so I'm setting a reminder to review it on Monday 😄

alexakreizinger commented on pipeline/inputs/kafka.md:
```diff
@@ -136,73 +136,93 @@ The example can be executed locally with `make start` in the `examples/kafka_fil
 ## AWS MSK IAM authentication
 
-Fluent Bit v4.0.4 and later supports authentication to Amazon MSK (Managed Streaming for Apache Kafka) clusters using AWS IAM. This lets you securely connect to MSK brokers with AWS credentials, leveraging IAM roles and policies for access control.
+Starting with version 4.0.4, Fluent Bit supports AWS IAM authentication for Amazon MSK clusters. This allows you to use your AWS credentials and IAM policies to control access to Kafka topics.
```
```suggestion
In Fluent Bit version 4.0.4 and later, you can use AWS IAM authentication for Amazon MSK clusters. This lets you use your AWS credentials and IAM policies to control access to Kafka topics.
```

alexakreizinger commented on pipeline/inputs/kafka.md:
```diff
@@ -136,73 +136,93 @@ The example can be executed locally with `make start` in the `examples/kafka_fil
+2. AWS credentials file (`~/.aws/credentials`)
+3. IAM instance profile (recommended for EC2)
+4. IAM task role (recommended for ECS)
+5. IAM service account (recommended for EKS)
```
```suggestion
1. Environment variables (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`)
1. AWS credentials file (`~/.aws/credentials`)
1. IAM instance profile (recommended for EC2)
1. IAM task role (recommended for ECS)
1. IAM service account (recommended for EKS)
```

alexakreizinger commented on pipeline/inputs/kafka.md:
```diff
@@ -136,73 +136,93 @@ The example can be executed locally with `make start` in the `examples/kafka_fil
+5. IAM service account (recommended for EKS)
 
-{% endhint %}
+### Required IAM permissions
```
```suggestion
### Required IAM permissions

{% hint style="info" %}

For detailed IAM policy configuration, consult your AWS administrator or refer to the [AWS MSK documentation](https://docs.aws.amazon.com/msk/latest/developerguide/iam-access-control.html).

{% endhint %}
```

alexakreizinger commented on pipeline/outputs/kafka.md:
```diff
@@ -237,83 +237,141 @@ pipeline:
 ## AWS MSK IAM authentication
 
-Fluent Bit 4.0.4 and later supports authentication to Amazon MSK (Managed Streaming for Apache Kafka) clusters using AWS IAM for the Kafka output plugin. This lets you securely send data to MSK brokers with AWS credentials, leveraging IAM roles and policies for access control.
+Starting with version 4.0.4, Fluent Bit supports AWS IAM authentication for Amazon MSK clusters. This allows you to use your AWS credentials and IAM policies to control access to Kafka topics.
```
```suggestion
In Fluent Bit version 4.0.4 and later, you can use AWS IAM authentication for Amazon MSK clusters. This lets you use your AWS credentials and IAM policies to control access to Kafka topics.
```

alexakreizinger commented on pipeline/outputs/kafka.md:
```diff
@@ -237,83 +237,141 @@ pipeline:
-If you are compiling Fluent Bit from source, ensure the following requirements are met to enable AWS MSK IAM support:
+- Access to an AWS MSK cluster with IAM authentication enabled
+- Valid AWS credentials (IAM role, access keys, or instance profile)
+- Network connectivity to your MSK brokers
```
```suggestion
To use AWS MSK IAM authentication, you must meet these requirements:

- You must have access to an AWS MSK cluster with IAM authentication enabled.
- You must have valid AWS credentials (IAM role, access keys, or instance profile).
- You must have network connectivity to your MSK brokers.
```

alexakreizinger commented on pipeline/outputs/kafka.md:
```diff
@@ -237,83 +237,141 @@ pipeline:
-    - Environment variables (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`)
-    - AWS credentials file (`~/.aws/credentials`)
-    - Instance metadata service (IMDS)
+For most use cases, simply set `rdkafka.sasl.mechanism` to `aws_msk_iam`:
```
```suggestion
For most use cases, the only necessary configuration step is to set `rdkafka.sasl.mechanism` to `aws_msk_iam`:
```

alexakreizinger commented on pipeline/outputs/kafka.md:
```diff
@@ -237,83 +237,141 @@ pipeline:
+
+{% hint style="info" %}
+For detailed IAM policy configuration, consult your AWS administrator or refer to the [AWS MSK documentation](https://docs.aws.amazon.com/msk/latest/developerguide/iam-access-control.html).
+{% endhint %}
```
```suggestion
```

alexakreizinger commented on pipeline/outputs/kafka.md:
```diff
@@ -237,83 +237,141 @@ pipeline:
+4. IAM task role (recommended for ECS)
+5. IAM service account (recommended for EKS)
+
+### Required IAM permissions
```
```suggestion
### Required IAM permissions

{% hint style="info" %}

For detailed IAM policy configuration, consult your AWS administrator or refer to the [AWS MSK documentation](https://docs.aws.amazon.com/msk/latest/developerguide/iam-access-control.html).

{% endhint %}
```

alexakreizinger reviewed (approved):
@kalavt thanks for opening a PR, and apologies for the delayed review—I'm making some suggestions in line with our style guide, and feel free to merge if everything looks good to you :)

alexakreizinger committed:
Apply suggestions from code review

Signed-off-by: Alexa Kreizinger <alexakreizinger@gmail.com>

eschabell committed:
Merge branch 'master' into feature/aws-msk-iam-auth

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
docs: pipeline: outputs: kafka: Fixed missing parameter.

  Applies to #2294

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/outputs/kafka.md:
```diff
@@ -248,82 +248,161 @@ pipeline:
+{% hint style="info" %}
+
+When using `aws_msk_iam`, Fluent Bit automatically sets `rdkafka.security.protocol` to `SASL_SSL`. You don't need to configure it manually.
 
```
@kalavt this would need to be addressed to pass the review please?

eschabell replied:
Fixed.

eschabell replied:
Fixed.

eschabell committed:
docs: pipeline: outputs: kafka: update MSK IAM parameter references

  - Replace aws_msk_iam/aws_msk_iam_cluster_arn references with
    rdkafka.sasl.mechanism and aws_region per PR fluent/fluent-bit#11270
  - Reword hint to reflect that setting rdkafka.sasl.mechanism=aws_msk_iam
    triggers automatic rdkafka.security.protocol=SASL_SSL

  Applies to #2294

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell reviewed (approved):
@kalavt waiting only on code PR merging.
